agent/grpc/resolver/resolver.go

```diff
@@ -141,12 +129,16 @@ func (s *ServerResolverBuilder) AddServer(server *metadata.Server) {
 	}
 }
 
+func uniqueID(server *metadata.Server) string {
+	return server.Datacenter + "-" + server.ID
+}
```

Comment:
Is it worth commenting the rationale for this function? The comment above maybe was too pedantic in that server node IDs kinda need to be unique across datacenters already IIRC to avoid issues with WAN gossip flapping since they are all in the same gossip pool. I don't specifically recall if it's a hard requirement though since raft/serf/catalog all treat node IDs a bit differently.

In the normal case they are all random UUIDs so are always unique anyway, it is just about possible that an operator could have chosen to override them all with `-node-id` and chosen to set them all to something the same etc.

In other words, I _think_ in all but the most bizarre edge case I can imagine, they are already globally unique in the cluster, I'm just not 100% sure that it's actually impossible to get things working with duplicated ones if you really try, it's certainly never advisable or reasonable too.

All that said, this is a simple way to ensure it's never not true so its fine to keep but it might puzzle future us as to why it's necessary so a comment might help?

Reply:
Good idea! I'll add a comment explaining the rational here